### PR TITLE
Add milestone tracking automation (mutation disabled)

### PR DESCRIPTION
**Pull request by sharwell**
Implements the following rule, but does not enable the mutation at this point.

MicrosoftTeams-image

### CODE REVIEW

File: .github/workflows/milestone-tracking.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Milestone tracking
+on:
+  pull_request:
+    types: [closed]
+jobs:
+  add_milestone:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'dotnet/roslyn' && github.event.pull_request.merged_at != null && github.event.pull_request.milestone == null && (github.event.pull_request.base.ref == 'main' || github.event.pull_request.base.ref == 'main-vs-deps') }}
```
Comment (jmarolf) [inline .github/workflows/milestone-tracking.yml:8]: we don't need the `main-vs-deps` statement for this repo

File: .github/workflows/milestone-tracking.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          query($org: String!, $repo: String!, $milestone: String!) {
+            repository(name: $repo, owner: $org) {
+              milestones(query: $milestone, first: 2) {
+                nodes {
+                  id
+                  title
+                }
+              }
+            }
+          }' -f org=$ORGANIZATION -f repo=$REPOSITORY -f milestone="$MILESTONE_NAME" > milestone_data.json
```
Comment (jmarolf) [inline .github/workflows/milestone-tracking.yml:26]: where is `MILESTONE_NAME` going to be defined?

Comment (sharwell) [reply .github/workflows/milestone-tracking.yml:26]: Line 14 above

File: .github/workflows/milestone-tracking.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    - name: Get milestone data
+      env:
+        ORGANIZATION: dotnet
+        REPOSITORY: roslyn-analyzers
+        MILESTONE_NAME: vNext
+      run: |
+        gh api graphql -f query='
+          query($org: String!, $repo: String!, $milestone: String!) {
+            repository(name: $repo, owner: $org) {
+              milestones(query: $milestone, first: 2) {
```
Comment (jmarolf) [inline .github/workflows/milestone-tracking.yml:19]: why `first: 2` shouldn't there only be one milestone with a matching name?

Comment (jmarolf) [reply .github/workflows/milestone-tracking.yml:19]: ah, its the first 2 nodes

**Review by jmarolf — APPROVED** [review]

File: .github/workflows/milestone-tracking.yml (hunk 3 above)
Comment (sharwell) [reply .github/workflows/milestone-tracking.yml:19]: I wanted to use the equivalent of `.Single()` for validation, but couldn't find it. The query appears to be a substring match, and the docs aren't clear about how to be more specific.

Comment (sharwell) [reply .github/workflows/milestone-tracking.yml:19]: ➡️ Validation is now added

**Pull request merged**